### PR TITLE
fix: selftouching polygon

### PR DESCRIPTION
**Pull request by emalfroy**

### CODE REVIEW

File: src/BuildingRegistry/Building/Building.cs [hunk 1]
```diff
@@ -41,7 +42,7 @@ public sealed partial class Building : AggregateRootEntity, ISnapshotable
         {
             var geometry = WKBReaderFactory.Create().Read(extendedWkbGeometry);
 
-            GuardPolygon(geometry);
+            GuardPolygon(geometry, isMeasured: false);
```
Comment (emalfroy) [inline src/BuildingRegistry/Building/Building.cs:45]: Plan was using the `IsSelfTouchingRingFormingHoleValid = true` thefore it didn't throw an exception with invalid selftouching polygon

File: src/BuildingRegistry/Legacy/Building.cs [hunk 2]
```diff
@@ -409,7 +409,7 @@ private static bool IsGeometryValid(ExtendedWkbGeometry validGeometry)
             }
 
             var geometry = WkbReader.Read(validGeometry);
-            return GeometryValidator.IsValid(geometry);
+            return GeometryValidator.IsValidMeasured(geometry);
```
Comment (jvandaal) [inline src/BuildingRegistry/Legacy/Building.cs:412]: Don't change legacy.

Comment (emalfroy) [reply src/BuildingRegistry/Legacy/Building.cs]: @ArneD changed it to IsValidOutlined. Will now fail on selftouching polygon.
or do we copy the GeometryValidator for legacy and leave it as it was?

Comment (ArneD) [reply src/BuildingRegistry/Legacy/Building.cs]: IsValidMeasured is same logic as before, so I think that's correct not outlined

File: src/BuildingRegistry/Building/Building_Grb.cs [hunk 3]
```diff
@@ -172,6 +172,8 @@ public void ChangeMeasurement(ExtendedWkbGeometry extendedWkbGeometry, BuildingG
                 throw new BuildingHasInvalidGeometryMethodException();
             }
 
+            GuardPolygon(WKBReaderFactory.Create().Read(extendedWkbGeometry));
```
Comment (ArneD) [inline src/BuildingRegistry/Building/Building_Grb.cs:175]: good catch

**Pull request merged**